### PR TITLE
[Enhancement](HttpServer) Support https interface

### LINKED ISSUE #15658: [Enhancement] add http interface authentication and support https
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, doris supports access to fe / be only through http protocol, not https protocol. In addition, http does not perform interface authentication, which may cause security problems.

### Solution

Add http interface authentication and support https for doris, eliminate potential security risks and strengthen enterprise-level security support.
[Support https protocol and interface authentication](https://cwiki.apache.org/confluence/display/DORIS/DSIP-029%3A+Support+https+protocol+and+interface+authentication)

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)]

### PR DESCRIPTION
**Pull request by yongjinhou**
# Proposed changes

Issue Number: close #15658

## Problem summary

1. Organize http documents
2. Add http interface authentication for FE
3. **Support https interface for FE**
4. Provide authentication interface
5. Add http interface authentication for BE
6. Support https interface for BE

## Checklist(Required)

* [x] Does it affect the original behavior
* [ ] Has unit tests been added
* [x] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -344,6 +344,35 @@ public class Config extends ConfigBase {
...
+     * ssl key store alias. "doris_ssl_certificate" by default.
+     */
+    @ConfField public static String key_store_alias = "doris_ssl_certificate";
+
+    /**
+     * ssl enable flag. false by default.
+     * If the value is false, http is supported. Otherwise, https is supported.
+     * Currently doris uses many ports, so http and https are not supported at the same time.
+     */
+    @ConfField public static boolean ssl_enable = false;
```
Comment (morningman) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:374]: Please update the document and describe how to use these config
https://doris.apache.org/zh-CN/docs/dev/admin-manual/config/fe-config

Comment (yongjinhou) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:374]: ok

Comment (yangzhg) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:374]:
```suggestion
    @ConfField public static boolean enable_https = false;
```

Comment (yongjinhou) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java]: ok

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
LocalTest is being tested.

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p1

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/UploadAction.java [hunk 2]
```diff
@@ -75,6 +77,13 @@ public Object upload(
             @PathVariable(value = TABLE_KEY) String tblName,
             @RequestParam("file") MultipartFile file,
             HttpServletRequest request, HttpServletResponse response) {
+        if (Config.enable_https && request.getServerPort() == Config.http_port) {
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/UploadAction.java:80]: extract this as a function ？

Comment (yongjinhou) [reply fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/UploadAction.java:80]: 好的

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 3]
```diff
@@ -338,16 +338,51 @@ public class Config extends ConfigBase {
...
      */
     @ConfField public static int http_port = 8030;
 
     /**
-     * Whether to enable all http interface authentication
+     * Fe https port
+     * Currently, all FEs' https port must be same.
      */
-    @ConfField public static boolean enable_all_http_auth = false;
+    @ConfField public static int https_port = 8050;
```
Comment (yiguolei) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:356]: why need a seperate https port? I think https just on the original http port, and maybe it is called enable_https.

**Review by yiguolei — CHANGES_REQUESTED** [review]
Not add seperate https port.

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java (hunk 3 above)
Comment (yongjinhou) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:356]: Current version of springboot does not support http and https sharing same port.

Comment (yongjinhou) [reply fe/fe-common/src/main/java/org/apache/doris/common/Config.java:356]: enable_all_http_auth and enable_https are two different parameters.

**Review by morningman — COMMENTED** [review]
I think it is ok to use 2 ports now.
But we should only enable one of them at same time.
And user can choose to use which one.

**Comment by yongjinhou** [comment]
> I think it is ok to use 2 ports now. But we should only enable one of them at same time. And user can choose to use which one.

Currently, if enable_https is false, only http is supported. If enable_https is true, both http and https are supported and http is redirected to https. This ensures that the previous link is still available, and that users, promethous, spark/flink-doris-connector, etc., can access using http without changing the protocol used.

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

**Comment by yongjinhou** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**